src/vmm/src/rpc_interface.rs

```diff
@@ -427,10 +464,32 @@ impl RuntimeApiController {
     }
 
     #[cfg(target_arch = "x86_64")]
-    fn create_snapshot(&mut self, params: CreateSnapshotParams) -> ActionResult {
+    fn create_snapshot(&mut self, create_params: &CreateSnapshotParams) -> ActionResult {
         let mut locked_vmm = self.vmm.lock().unwrap();
-        persist::create_snapshot(&mut locked_vmm, params, VERSION_MAP.clone())
-            .map_err(VmmActionError::CreateSnapshot)
+        let create_start_us = utils::time::get_time_us(utils::time::ClockType::Monotonic);
+
+        persist::create_snapshot(&mut locked_vmm, create_params, VERSION_MAP.clone())
+            .map_err(VmmActionError::CreateSnapshot)?;
+
+        match create_params.snapshot_type {
+            SnapshotType::Full => {
+                update_metric_with_elapsed_time(
+                    &METRICS.latencies_us.vmm_full_create_snapshot,
+                    create_start_us,
+                    "creating the full snapshot",
```

Comment:
Hmmm, now "create_snapshot()" as action string would be bad here as we also need to account for inc/full. So how about "create_snapshot(diff/full)" ?